src/builtins/BuiltinTemporal.cpp

```diff
@@ -642,7 +768,7 @@ static Value builtinTemporalCalendarPrototypeYear(ExecutionState& state, Value t
     Value calendar = thisValue;
     ASSERT(calendar.asObject()->asTemporalCalendarObject()->getIdentifier()->equals("iso8601"));
     Value temporalDateLike = argv[0];
-    if (!(temporalDateLike.isObject() && (temporalDateLike.asObject()->isTemporalPlainDateObject() || temporalDateLike.asObject()->isTemporalPlainDateTimeObject() || temporalDateLike.asObject()->isTemporalYearMonthObject()))) {
+    if (!(temporalDateLike.isObject() && (temporalDateLike.asObject()->isTemporalPlainDateObject() || temporalDateLike.asObject()->isTemporalPlainDateTimeObject() || temporalDateLike.asObject()->isTemporalPlainYearMonthObject()))) {
```

Comment:
There are a lot of similar big checks. Maybe we could introduce a helper function for them.